master/internal/command/command_manager.go

```diff
@@ -1,81 +1,102 @@
-// Package command provides utilities for commands.
-//
-//nolint:dupl
 package command
 
 import (
-	"net/http"
-
-	"github.com/labstack/echo/v4"
-
-	"github.com/determined-ai/determined/master/internal/api"
-	"github.com/determined-ai/determined/master/internal/db"
-	"github.com/determined-ai/determined/master/internal/rm"
-	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/internal/task"
 	"github.com/determined-ai/determined/master/pkg/model"
-	"github.com/determined-ai/determined/master/pkg/tasks"
+	"github.com/determined-ai/determined/master/pkg/protoutils"
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 	"github.com/determined-ai/determined/proto/pkg/commandv1"
 )
 
-// CreateGeneric is a request to managers to create a generic command.
-type CreateGeneric struct {
-	ContextDirectory []byte
-	Spec             *tasks.GenericCommandSpec
+func (cs *commandService) GetCommands(req *apiv1.GetCommandsRequest) (*apiv1.GetCommandsResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	resp := &apiv1.GetCommandsResponse{}
+	users := make(map[string]bool, len(req.Users))
+	for _, user := range req.Users {
+		users[user] = true
+	}
+	userIds := make(map[int32]bool, len(req.UserIds))
+	for _, user := range req.UserIds {
+		userIds[user] = true
+	}
+	for _, c := range cs.getAllNTSCType(model.TaskTypeCommand) {
+		cmd := c.toCommand()
+		if (len(users) == 0 && len(userIds) == 0) || users[cmd.Username] || userIds[cmd.UserId] {
+			resp.Commands = append(resp.Commands, cmd)
+		}
+	}
+	return resp, nil
 }
 
-type commandManager struct {
-	db *db.PgDB
-	rm rm.ResourceManager
+func (cs *commandService) GetCommand(req *apiv1.GetCommandRequest) (*apiv1.GetCommandResponse, error) {
+	cs.mu.Lock()
+	defer cs.mu.Unlock()
+
+	c, err := cs.getNTSC(model.TaskID(req.CommandId), model.TaskTypeCommand)
+	if err != nil {
+		cs.syslog.Errorf("could not get notebook: %s", req.CommandId)
```

Comment:
return the error dont log here and in a lot of places throughout this file (we're calling .toCommand on nil or zero value).